autopush/endpoint.py

```diff
@@ -254,6 +278,58 @@ def _db_error_handling(self, d):
         d.addErrback(self._response_err)
         return d
 
+    def _store_auth(self, jwt, crypto_key, token, result):
+        if jwt.get('exp', 0) < time.time():
+            raise VapidAuthException("Invalid bearer token: Auth expired")
+        jwt['crypto_key'] = crypto_key
+        jwt['token'] = token
+        self._client_info['jwt'] = jwt
+        return result
+
+    def _invalid_auth(self, fail):
+        if isinstance(fail.value, VapidAuthException):
+            raise fail.value
+        message = fail.value.message or repr(fail.value)
+        if isinstance(fail.value, AssertionError):
+            message = "A decryption error occurred"
+        log.msg("Invalid bearer token: " + message, **self._client_info)
+        raise VapidAuthException("Invalid bearer token: " + message)
+
+    def _process_auth(self, result):
+        """Process the optional VAPID auth token.
+
+        VAPID requires two headers to be present;
+        `Authorization: bearer ...` and `Crypto-Key: p256ecdsa=..`.
+        The problem is that VAPID is optional and Crypto-Key can carry
+        content for other functions."""
```

Comment:
nit: triple quote needs a line break drop, no line break after it before the code starts